src/App.jsx

```diff
@@ -1,56 +1,94 @@
 import "./App.css"
+import React, { useState } from 'react';
 
 function App() {
+  const [firstPanelNum, updateNum1] = useState(0);
+  const [secondPanelNum, updateNum2] = useState(0);
+  const [operator, setOperator] = useState('+');
+  const [result, setResult] = useState(0);
+
+  const handleClickPanel1 = (num) => {
+    updateNum1(num);
+  };
+
+  const handleClickPanel2 = (num) => {
+    updateNum2(num)
+  };
+
+  const handleClickOperator = (operator) => {
+    setOperator(operator)
+  };
+
+  const handleEquals = () => {
+    let res;
+    switch (operator) {
+      case '+':
+        res = firstPanelNum + secondPanelNum;
+        break;
+      case '-':
+        res = firstPanelNum - secondPanelNum;
+        break;
+      case '*':
+        res = firstPanelNum * secondPanelNum;
+        break;
+      case '÷':
+        res = firstPanelNum / secondPanelNum;
+        break;
+      default:
+        res = 0;
+    }
+    setResult(res);
+  };
 
   return (
     <div className="calculator">
       <div className="panel">
-        <p>0</p>
+        <p>{firstPanelNum}</p>
         <div className="numbers">
-          <button>1</button>
-          <button>2</button>
-          <button>3</button>
-          <button>4</button>
-          <button>5</button>
-          <button>6</button>
-          <button>7</button>
-          <button>8</button>
-          <button>9</button>
-          <button>0</button>
-          <button>Clear</button>
+          <button onClick={() => handleClickPanel1(1)}>1</button>
```

Comment:
see if you can use map on an array to generate all these buttons

Reply:
Yes, Done!